src/main/java/com/faforever/api/user/SteamService.java

```diff
@@ -73,15 +80,46 @@ void validateSteamRedirect(HttpServletRequest request) {
     String recodedUri = builder.toUriString().replace("+", "%2B");
     log.debug("Verification uri: {}", recodedUri);
 
-    // the Spring RestTemplate still struggles with the + character so we use the default Java http client
+    // the Spring RestTemplate still struggles with the + character, so we use the default Java http client
     String result = HttpClient.newHttpClient()
       .send(HttpRequest.newBuilder(new URI(recodedUri)).build(), BodyHandlers.ofString())
       .body();
 
     if (result == null || !result.contains("is_valid:true")) {
-      throw new IllegalArgumentException("Steam redirect could not be validated! Original response:\n" + result);
+      handleInvalidOpenIdRedirect(request, result);
     } else {
       log.debug("Steam response successfully validated.");
     }
   }
+
+  void handleInvalidOpenIdRedirect(final HttpServletRequest request, final String openIdResponseBody) {
+    boolean containsIdentityParam = request.getParameterMap().containsKey("openid.identity");
+    final String steamId;
+
+    if (containsIdentityParam)
+    {
+       steamId = parseSteamIdFromLoginRedirect(request);
+    } else {
```

Comment:
The `containsIdentityParam` check could be moved into `parseSteamIdFromLoginRedirect` which would then return an Optional.
Then here we could just `.orElseThrow()`